app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
+        if params[:next_of_kin_name].present? && params[:next_of_kin_email].present?
+          if @user.fannish_next_of_kin.nil?
+            @user.fannish_next_of_kin = FannishNextOfKin.new(:user_id => params[:user_login],
+              :kin_id => submitted_kin_user.id,
+              :kin_email => params[:next_of_kin_email])
+            success_message << ts('Fannish next of kin added.')
           else
-            @user.suspended = true
-            @suspension_days = params[:suspend_days].to_i
-            @user.suspended_until = @suspension_days.days.from_now
-            if @user.save && @user.suspended? && !@user.suspended_until.blank?
-              @user.create_log_item( options = {:action => ArchiveConfig.ACTION_SUSPEND, :note => @admin_note, :admin_id => current_admin.id, :enddate => @user.suspended_until})
-              flash[:notice] = ts("User has been temporarily suspended")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            else
-              flash[:error] = ts("User could not be suspended")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            end
+            @user.fannish_next_of_kin = @user.fannish_next_of_kin
           end
-        elsif params[:admin_action] == 'ban'
+        end
+
+        # update the next of kin user if the field is present and changed
+        if params[:next_of_kin_name].present? && !(submitted_kin_user.id == @user.fannish_next_of_kin.kin_id)
+          @user.fannish_next_of_kin.kin_id = submitted_kin_user.id
+          success_message << ts('Fannish next of kin user updated.')
+        end
+
+        # update the next of kin email is the field is present and changed
+        if params[:next_of_kin_email].present? && !(params[:next_of_kin_email] == @user.fannish_next_of_kin.kin_email)
+          @user.fannish_next_of_kin.kin_email = params[:next_of_kin_email]
+          success_message << ts('Fannish next of kin email updated.')
+        end
+        
+        # delete the next of kin if the fields are blank and changed
+        if params[:next_of_kin_user].blank? && params[:next_of_kin_email].blank? && @user.fannish_next_of_kin
+          @user.fannish_next_of_kin.destroy
+          success_message << ts('Fannish next of kin removed.')
+        end
+
+        # create warning
+        if params[:admin_action] == 'warn'
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_WARN, :note => @admin_note, :admin_id => current_admin.id})
+          success_message << ts('Warning was recorded.')
+        end
+
+        # create suspension
+        if params[:admin_action] == 'suspend'
+          @user.suspended = true
+          @suspension_days = params[:suspend_days].to_i
+          @user.suspended_until = @suspension_days.days.from_now
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_SUSPEND, :note => @admin_note, :admin_id => current_admin.id, :enddate => @user.suspended_until})
```

Comment:
Useless assignment to variable - `options`.<br>Line is too long. [173/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.<br>Space inside parentheses detected.